os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -314,89 +313,78 @@ static void up_dumpstate(struct tcb_s *fault_tcb, uint32_t asserted_location)
-			lldbg_noarg("Current SP is User Thread SP: %08x\n", sp);
-			lldbg_noarg("User stack:\n");
+			lldbg("Current SP is User Thread SP: %08x\n", sp);
+			lldbg("User stack:\n");
 		} else {
 			is_sp_corrupt = true;
 		}
 #endif
 	} else if ((sp <= stackbase) && (sp > (stackbase - stacksize))) {
-		lldbg_noarg("Current SP is User Thread SP: %08x\n", sp);
-		lldbg_noarg("User stack:\n");
+		lldbg("Current SP is User Thread SP: %08x\n", sp);
+		lldbg("User stack:\n");
 	} else {
 		is_sp_corrupt = true;
 	}
 
 	if (is_sp_corrupt) {
-		lldbg_noarg("ERROR: Stack pointer is not within the allocated stack\n");
-		lldbg_noarg("ERROR: SP = 0x%08x\n", sp);
+		lldbg("ERROR: Stack pointer is not within the allocated stack\n");
+		lldbg("ERROR: SP = 0x%08x\n", sp);
 		if (is_irq_assert) {
 #if CONFIG_ARCH_INTERRUPTSTACK > 7
-			lldbg_noarg("ERROR: IRQ stack = 0x%08x - 0x%08x\n", istackbase - istacksize + 1, istackbase);
+			lldbg("ERROR: IRQ stack = 0x%08x - 0x%08x\n", istackbase - istacksize + 1, istackbase);
 #else
-			lldbg_noarg("ERROR: USR stack = 0x%08x - 0x%08x\n", stackbase - stacksize + 1, stackbase);
+			lldbg("ERROR: USR stack = 0x%08x - 0x%08x\n", stackbase - stacksize + 1, stackbase);
 #endif
 		} else {
-			lldbg_noarg("ERROR: USR stack = 0x%08x - 0x%08x\n", stackbase - stacksize + 1, stackbase);
+			lldbg("ERROR: USR stack = 0x%08x - 0x%08x\n", stackbase - stacksize + 1, stackbase);
 		}
 		
-		lldbg_noarg("Wrong Stack pointer %08x: %08x %08x %08x %08x %08x %08x %08x %08x\n",
+		lldbg("Wrong Stack pointer %08x: %08x %08x %08x %08x %08x %08x %08x %08x\n",
 		sp, *((uint32_t *)sp + 0), *((uint32_t *)sp + 1), *((uint32_t *)sp + 2), ((uint32_t *)sp + 3),
 		*((uint32_t *)sp + 4), ((uint32_t *)sp + 5), ((uint32_t *)sp + 6), ((uint32_t *)sp + 7));
-
 		if (is_irq_assert) {
 #if CONFIG_ARCH_INTERRUPTSTACK > 7
-			lldbg_noarg("IRQ stack dump:\n");
+			lldbg("IRQ stack dump:\n");
 			up_stackdump(istackbase - istacksize + 1, istackbase);
 #else
-			lldbg_noarg("User thread stack dump:\n");
+			lldbg("User thread stack dump:\n");
 			up_stackdump(stackbase - stacksize + 1, stackbase);
 #endif
 		} else {
-			lldbg_noarg("User thread stack dump:\n");
+			lldbg("User thread stack dump:\n");
 			up_stackdump(stackbase - stacksize + 1, stackbase);
 		}
 	} else {
 		/* Dump the stack region which contains the current stack pointer */
-		lldbg_noarg("  base: %08x\n", stackbase);
-		lldbg_noarg("  size: %08x\n", stacksize);
+		lldbg("  base: %08x\n", stackbase);
+		lldbg("  size: %08x\n", stacksize);
 #ifdef CONFIG_STACK_COLORATION
-		lldbg_noarg("  used: %08x\n", up_check_assertstack((uintptr_t)(stackbase - stacksize), stackbase));
+		lldbg("  used: %08x\n", up_check_assertstack((uintptr_t)(stackbase - stacksize), stackbase));
 #endif
 		up_stackdump(sp, stackbase);
 	}
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("\n");
+
+	/* Dump the registers */
+	arm_registerdump(CURRENT_REGS);
```

Comment:
You split the register dump as a separate function. How about using the same way for stack and irq?

Reply:
The `up_dumpstate` function prints 
1. irq
2. stack
3. register
4. and so on

It does 1st and 2nd in the function but does 3rd by calling separate function.
I suggest using the same way for above. Let's separate 1st and 2nd and call them from the up_dumpstate.